UnitTest/Logic/SprintLogicTest.cs

```diff
@@ -0,0 +1,191 @@
+                .Returns(true);
+
+            mockSprintEngine.Setup(x => x.SprintExists(sprint.SprintId))
+               .Returns(false);
+
+            SprintLogic sprintLogic = new SprintLogic(mockSprintEngine.Object);
+
+            Action act = () => { sprintLogic.EditSprint(sprint); };
+            act.Should().Throw<Exception>()
+                .And.Message
+                .Should().Be("Sprint does not exists");
+
+            mockSprintEngine.Verify(x => x.EditSprint(sprint), Times.Never);
+        }
+
+        [Fact]
+        public void GetSprintById_ReturnsSuccessfull()
+        {
+            var sprint = CreateSampleSprint();
+            mockSprintEngine.Setup(x => x.GetSprint(1))
+               .Returns(sprint);
+
+            var expected = CreateSampleSprint();
+
+            SprintLogic SprintLogic = new SprintLogic(mockSprintEngine.Object);
+
+            var actual = SprintLogic.GetSprint(1);
+
+            actual.Should().BeEquivalentTo(expected);
+        }
+
+        [Fact]
+        public void DeleteSprint_SprintDoesNotExists_ThrowsException()
+        {
+            var sprint = CreateSampleSprint();
+            mockSprintEngine.Setup(x => x.RemoveSprint(sprint))
+                .Returns(true);
+
+            mockSprintEngine.Setup(x => x.GetSprint(1))
+                .Returns((Sprint)null);
+
+            SprintLogic SprintLogic = new SprintLogic(mockSprintEngine.Object);
+            Action act = () => { SprintLogic.RemoveSprint(sprint.SprintId); };
+
+            act.Should().Throw<Exception>()
+                .And.Message
+                .Should().Be("Sprint does not exists");
+
+            mockSprintEngine.Verify(x => x.RemoveSprint(sprint), Times.Never);
+        }
+
+        [Fact]
+        public void DeleteSprint_SprintDeleted_ReturnsSuccessfull()
+        {
+            var sprint = CreateSampleSprint();
+            mockSprintEngine.Setup(x => x.RemoveSprint(sprint))
+                           .Returns(true);
+
+            mockSprintEngine.Setup(x => x.GetSprint(sprint.SprintId))
+                .Returns(sprint);
+
+            SprintLogic SprintLogic = new SprintLogic(mockSprintEngine.Object);
+
+            bool expected = true;
+            bool actual = SprintLogic.RemoveSprint(sprint.SprintId);
+
+            Assert.Equal(expected, actual);
+
+            mockSprintEngine.Verify(x => x.RemoveSprint(sprint), Times.Once);
+        }
+
+        [Fact]
+        public void GetSprints_ReturnsSuccessfull()
+        {
+            var sprint1 = CreateSampleSprint();
+            Sprint sprint2 = new Sprint()
+            {
+                SprintId = 2,
+                SprintName = "test2",
+                SprintPoints = 2.0m,
+                StartDate = new DateTime(2019, 6, 1),
+                EndDate = new DateTime(2019, 12, 1),
+                SprintStatusId = 1,
+                CreatedBy = "user",
+                CreatedDate = new DateTime(2019, 12, 30)
+            };

```

Comment:
opportunity to reuse code. Try using default parameter in case you need SprintId to be unique